utils/posts.js

```diff
@@ -41,15 +43,68 @@ export function getPostMetadata(id) {
             ...matterResult.data,
         },
     };
-
+    
     const title = post.meta.title;
-    var seoTitle = title.toLowerCase().replace(/[^a-z0-9]+/g, ' ').trim().replace(/ /g, '-');
-
+    const seoTitle = title.toLowerCase().replace(/[^a-z0-9]+/g, ' ').trim().replace(/ /g, '-');
+    
     post.meta.seoTitle = seoTitle;
 
+    var tags = post.meta.tags;
+    if(tags) {
+        tags = tags.split(',').map(tag => tag.trim());
+	tags = tags.map(tag => ({
+            content: tag,
+            id: tag.replace(/ /g, '-').toLowerCase(),
+	}));
+    } else {
+        tags = null
+    }
+    post.tags = tags;
     return post;
 }
 
+// Retrieves a list of all unique tags
+export function getTags() {
+    const posts = getPosts();
+    var tags = [];
+    for(var i = 0; i < posts.length; i++) {
+        if(posts[i].tags) {
+            tags = tags.concat(posts[i].tags);
+        };
+    }
+    const uniqueSorted = Array.from(new Set(tags.map(tag => tag.id))).map(id => {
+        return tags.find(tag => tag.id === id);
+    }).sort((a, b) => {
+        const result = a.id > b.id ? 1 : -1;
+	return result;
+    });
+    return uniqueSorted;
+}
+
+//Retrieves the content of a specific tag
+export function getTagContent(id) {
+    return getTags().find(tag => tag.id === id).content;
+}
+
+// Retrieves all posts sorted into their respective tag(s)
+export function getTagSortedPosts() {
+    const posts = getPosts();
+    const tags = getTags();
+    var sortedPosts = {};
+    for(var i = 0; i < tags.length; i++) {
```

Comment:
We're mixing different styles of iteration.

Some places, we are doing `.map()`, but other places, we are doing `for(var i=0; i < stuff.length; ++i)`

Reply:
@jontsai I'm using a `for` loop because `map` cannot map to an object.

Reply:
@jontsai I'm using `forEach`.